lib/caching/NonceCache.ts

```diff
@@ -0,0 +1,184 @@
+import { NonceType } from '../interfaces/index';
+import { Query } from '../apis/query';
+import { CachedNonceInfo, INonceCacheProvider, NonceData } from './index';
+import { AccountUtils, Utils } from '../utils';
+import log from 'loglevel';
+
+const TX_PROCESSING_TIME_MS = 120000;
+const NONCE_LOCK_POLL_INTERVAL_MS = 500;
+const MAX_NONCE_LOCK_TIME_MS = TX_PROCESSING_TIME_MS + NONCE_LOCK_POLL_INTERVAL_MS;
+const EXPIRY_UPDATE_ENUM = {
+  DoNotUpade: false,
+  UpdateExpiry: true
+};
+
+export class NonceCache {
+  private cacheProvider: INonceCacheProvider;
+
+  constructor(cacheProvider: INonceCacheProvider) {
+    this.cacheProvider = cacheProvider;
+  }
+
+  public async init() {
+    this.cacheProvider = await this.cacheProvider.connect();
+  }
+
+  public async setNonceCacheForUserIfRequired(signerAddress: string) {
+    signerAddress = AccountUtils.convertToPublicKeyIfNeeded(signerAddress);
```

Comment:
Should we create a different variable so we don't modify the given parameter?

Reply:
I think it makes sense to replace the same variable because we are not changing the meaning of it, just its format.